modules/conversantBidAdapter.js

```diff
@@ -27,6 +27,118 @@ const BIDDER_CODE = 'conversant';
 export const storage = getStorageManager({gvlid: GVLID, bidderCode: BIDDER_CODE});
 const URL = 'https://web.hb.ad.cpe.dotomi.com/cvx/client/hb/ortb/25';
 
+function setGDPR(bidderRequest, request) {
+  // Add GDPR flag and consent string
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(request, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+
+    if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+      deepSetValue(request, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
+    }
+  }
+
+  if (bidderRequest.uspConsent) {
+    deepSetValue(request, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+}
+
+function setSiteId(bidRequest, request) {
+  if (bidRequest.params.site_id) {
+    if (request.site) {
+      request.site.id = bidRequest.params.site_id;
+    }
+    if (request.app) {
+      request.app.id = bidRequest.params.site_id;
+    }
+  }
+}
+
+function setPubcid(bidRequest, request) {
+  // Add common id if available
+  const pubcid = getPubcid(bidRequest);
+  if (pubcid) {
+    deepSetValue(request, 'user.ext.fpc', pubcid);
+  }
+}
+
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  request: function (buildRequest, imps, bidderRequest, context) {
+    const request = buildRequest(imps, bidderRequest, context);
+    request.at = 1;
+    request.cur = 'USD';
+    if (context.bidRequests) {
+      const bidRequest = context.bidRequests[0];
+      setSiteId(bidRequest, request);
+      setPubcid(bidRequest, request);
+    }
+
+    if (bidderRequest) {
+      setGDPR(bidderRequest, request);
+    }
+
+    return request;
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    const data = {
+      secure: 1,
+      bidfloor: getBidFloor(bidRequest) || 0,
+      displaymanager: 'Prebid.js',
+      displaymanagerver: '$prebid.version$'
+    };
+    copyOptProperty(bidRequest.params.tag_id, data, 'tagid');
+    mergeDeep(imp, data, imp);
+    return imp;
+  },
+  bidResponse: function (buildBidResponse, bid, context) {
+    if (!bid.price) return;
+    if (!context.mediaType && context.bidRequest.mediaTypes) {
```

Comment:
Yes, context is built differently.  it's created inside of toORTB and from ORTB but within them they reference the original request.

I don't want to check if bidRequest.mediaTypes is an array  because it shouldn't be.  Here all I want to know is if it's a banner, video, or native and make sure it's set in the context, otherwise it won't create the markup since it won't get past setResponseMediaType.  I couldn't have handled this in the request section but here I can ensure it's set whether it's in the response or not.

Reply:
If we trust it to be an array, why couldn't we trust it to be a valid type as well?

Reply:
It's not an array.  It would look like context: { mediaType: {banner: {}}} or context: { mediaType: {video: {}}}
and I need the literal string 'banner' or 'video' which I'm ripping out of the mediaType section since I'm not expecting it to be explicitly defined in the convex response.  And while that type is never explicitly validated, it's the key to creating the markup.  Like if mediaType isn't set to banner the adm and nurl wont be set because fillBannerResponse will be a NOP

Reply:
Ah ok.  Thanks for the clarification.